rpc/query_server.go

```diff
@@ -277,15 +277,17 @@ func (s *QueryServer) queryEvm(caller, contract loom.Address, query []byte) ([]b
 
 // https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_call
 func (s *QueryServer) EthCall(query eth.JsonTxCallObject, block eth.BlockHeight) (resp eth.Data, err error) {
+	snapshot := s.StateProvider.ReadOnlyState()
 	var caller loom.Address
 	if len(query.From) > 0 {
-		caller, err = s.getEthAccount(query.From)
+		caller, err = s.getEthAccount(snapshot, query.From)
 		if err != nil {
 			return resp, err
 		}
 	} else {
 		caller = loom.RootAddress(s.ChainID)
 	}
+	snapshot.Release()
```

Comment:
Replace this with a `defer snapshot.Release()` at the top of this function, and pass the snapshot into `s.queryEvm`. Probably will need to change `s.queryPlugin` to also take a snapshot as input.